pootle/apps/pootle_store/diff.py

```diff
@@ -0,0 +1,302 @@
+    def db_target(self):
+        return multistring_to_python(self.db_unit["target_f"])
+
+    @property
+    def context_changed(self):
+        return self.db_context != self.file_unit["context"]
+
+    @property
+    def developer_comment_changed(self):
+        return self.db_developer_comment != self.file_unit["developer_comment"]
+
+    @property
+    def file_locations(self):
+        return "\n".join(self.file_unit["locations"])
+
+    @property
+    def locations_changed(self):
+        return self.db_unit["locations"] != self.file_locations
+
+    @property
+    def source_changed(self):
+        return not ((self.file_unit["source"] == "" and self.db_source == "")
+                    or self.file_unit["source"] == self.db_source)
+
+    @property
+    def target_changed(self):
+        return not ((self.file_unit["target"] == "" and self.db_target == "")
+                    or self.file_unit["target"] == self.db_target)
+
+    @property
+    def translator_comment_changed(self):
+        return (self.db_translator_comment
+                != self.file_unit["translator_comment"])
+
+
+class StoreDiff(object):
+
+    def __init__(self, db_store, file_store, file_revision):
+        self.db_store = db_store
+        self.file_store = file_store
+        self.db_revision = db_store.get_max_unit_revision()
+        self.file_revision = file_revision
+
+    @cached_property
+    def active_units(self):
+        return [unitid for unitid, unit in self.db_units.items()
+                if unit['state'] != OBSOLETE]
+
+    @cached_property
+    def db_units(self):
+        """All of the db units regardless of state or revision"""
+        db_units = OrderedDict()
+        _db_units = self.db_store.unit_set.values("unitid", "state", "id",
+                                                  "index", "revision",
+                                                  "source_f", "target_f",
+                                                  "developer_comment",
+                                                  "translator_comment",
+                                                  "locations", "context")
+        for unit in _db_units:
+            db_units[unit["unitid"]] = unit
+        return db_units
+
+    @cached_property
+    def file_units(self):
+        file_units = OrderedDict()
+        for unit in self.file_store.units:
+            if not unit.isheader():
+                dev_comment = unit.getnotes(origin="developer")
+                translator_comment = unit.getnotes(origin="translator")
+                funit = {"unitid": unit.getid(),
+                         "context": unit.getcontext(),
+                         "locations": unit.getlocations(),
+                         "source": unit.source,
+                         "target": unit.target,
+                         "state": unit.get_state_n(),
+                         "developer_comment": dev_comment,
+                         "translator_comment": translator_comment}
```

Comment:
Can you please put it like:

```python
                file_units[unit.getid()] = {
                    'unitid': unit.getid(),
                    'context': unit.getcontext(),
                    'locations': unit.getlocations(),
                    'source': unit.source,
                    'target': unit.target,
                    'state': unit.get_state_n(),
                    'developer_comment': dev_comment,
                    'translator_comment': translator_comment,
                }
```

It is easier to read and doesn't use extra variables.